utils/global-state-update-gen/src/generic.rs

```diff
@@ -284,32 +289,93 @@ fn find_large_bids<T: StateReader>(
 fn create_or_update_bid<T: StateReader>(
     state: &mut StateTracker<T>,
     pub_key: &PublicKey,
-    stake: U512,
+    recipient: &SeigniorageRecipient,
 ) {
     if state
         .get_bids()
         .get(pub_key)
         .and_then(|bid| bid.total_staked_amount().ok())
-        == Some(stake)
+        == recipient.total_stake()
```

Comment:
If we change the delegators though, in such a way that the `total_stake()` doesn't change, then shouldn't we iterate over the delegators anyway?

Reply:
I can see it being fixed in [`fa6f877` (#3305)](https://github.com/casper-network/casper-node/pull/3305/commits/fa6f877509be5454cfa2c93a0db60a0d0ca9ca7d)